CONTRIBUTING.md

```diff
@@ -35,3 +39,164 @@ The development workflow is the following:
 - When changed are validated, your branch is merged into `master`
 - `master` is automatically tested on Travis
 - Docker builds a new `latest` image
+
+## Coding Style
+
+### Bash and Shell
+
+When refactoring, writing or altering Script, that is Shell and Bash scripts, in any way, adhere to these rules:
+
+1. **Adjust your style of coding to the style that is already present**! Even if you do not like it, this is due to consistency. Look up the GNU coding style guide. There was a lot of work involved in making these scripts consistent.
+2. **Use `shellcheck` to check your scripts**! Your contributions are checked by TravisCI with shellcheck.
+3. There is a **`.editorconfig`** file. Make your IDE use it or adhere to it manually!
+4. It's okay to use `/bin/bash` instead of `/bin/sh`. You can alternatively use `/usr/bin/env bash`.
+5. `setup.sh` provides a good starting point to look for.
+6. When appropriate, use the `set` builtin. We recommend `set -euEo pipefail` (very strong) or `set -uE` (weaker).
+
+#### Styling rules
+
+##### initial description
+
+When writing a script, provide the version and the script's task like so:
+
+``` BASH
+#!/usr/bin/env bash
+
+# version  0.1.0
+#
+# <TASK DESCRIPTION> -> cut this off
+# to make it not longer than approx.
+# 60 cols.
+```
+
+We use [semantic versioning](https://semver.org/) - so do you.
+
+##### if-else-statements
+
+``` BASH
+if <CONDITION1>
```

Comment:
I generally like the formatting cleanup, but I am personally not a fan of the verbose syntax when it comes to conditionals and functions (yes, I have seen that you already did so when updating start.sh).

What is your take on this @erik-wramner ?

Edit: Generally speaking I think its a very nice idea to add a style guide!

Reply:
I see where you are coming from:) Even though I was not asked specifically, I would like to respond too:

I'm coming from C and Rust where auto-formatters are frequently used and more robust. That's where I got the formatting from. I would like to offer a compromise here: I will look for a formatter and work out auto-formatting for it. I would really like function braces (`{...}`) to be on the next line, but I will make the `if-else-statement` less verbose again. Is that okay with you @fbartels ?

Reply:
Oh, don't get me wrong. I am not objecting to the syntax, just saying that it would not be my personal preference. In the end @erik-wramner is way more active on maintaining this project than I am, so I would give his opinion the higher weight.

Reply:
That's actually nice to hear. I am currently about trying to implement `shfmt, but I dislike it. I would rather like to stick to the current style. But as you said @fbartels, @erik-wramner should take precedence.

Reply:
I'm not a bash expert, I work mostly with Java and C# and before that C/C++. I generally try to match the style of the scripts as I find them. In short I don't have any strong views on this. I like compact scripts, but it is more important that they are correct.

I find the 60 character limit a bit short. I generally use 120 characters with Java. 80 should work everywhere?

In addition I don't think that I'm alone in being a bit short on bash skills, so I would actually encourage comments that describe what the code does when the author thinks it is needed. Plus if Travis verifies the formatting then the local build should do it as well...

Reply:
Alright. I resolved this with 77520bf

We'll stick to the majority of guidelines, but I changed 60 to 80 with respect to column width, and I changed the description/comment rules. My rationale for sticking to the rules is that with Bash correctness is way more important than compactness. Therefore I would like to embrace this style.